features/step_definitions/collection.js

```diff
@@ -0,0 +1,267 @@
+    }
+
+    return request({
+        uri: `${this.instance}/${this.namespace}/collections/${id}`,
+        method: 'DELETE',
+        auth: {
+            bearer: this.jwt
+        },
+        json: true
+    }).then((response) => {
+        Assert.oneOf(response.statusCode, [204, 404]);
+    });
+}
+
+defineSupportCode(({ Before, Given, Then, When, After }) => {
+    Before('@collections', function hook() {
+        this.repoOrg = this.testOrg;
+        this.repoName = 'functional-collections';
+        this.pipelineId = null;
+        this.firstCollectionId = null;
+        this.secondCollectionId = null;
+    });
+
+    When(/^they create a new collection "myCollection" with that pipeline$/, function step() {
+        return this.ensurePipelineExists({ repoName: this.repoName })
+            .then(() => {
+                const requestBody = {
+                    name: 'myCollection',
+                    pipelineIds: [this.pipelineId]
+                };
+
+                return createCollection.call(this, requestBody)
+                    .then((response) => {
+                        Assert.oneOf(response.statusCode, [409, 201]);
+
+                        if (response.statusCode === 201) {
+                            this.firstCollectionId = response.body.id;
+                        } else {
+                            const str = response.body.message;
+
+                            this.firstCollectionId = str.split(': ')[1];
+                        }
+                    });
+            });
+    });
+
+    Then(/^they can see that collection$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections/${this.firstCollectionId}`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.statusCode, 200);
+            Assert.strictEqual(response.body.name, 'myCollection');
+        });
+    });
+
+    Then(/^the collection contains that pipeline$/, function step() {
+        const pipelineId = parseInt(this.pipelineId, 10);
+
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections/${this.firstCollectionId}`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.statusCode, 200);
+            Assert.deepEqual(response.body.pipelineIds, [pipelineId]);
+        });
+    });
+
+    When(/^they create a new collection "myCollection"$/, function step() {
+        return createCollection.call(this, { name: 'myCollection' })
+            .then((response) => {
+                Assert.oneOf(response.statusCode, [409, 201]);
+
+                if (response.statusCode === 201) {
+                    this.firstCollectionId = response.body.id;
+                } else {
```

Comment:
This step should be asserting that "myCollection" is brand new and never been created before. Why is there an `else` case?

Reply:
So what is the difference between the cases "they create a new collection..." and "they have a collection..."?

Reply:
Agreed with @joelseq . When I read this scenario, I don't think it matters if they just created myCollection or it was created earlier, just as long as it exists in their list of collections